float32 x
float32 y
float32 z
float32 delta_x
float32 delta_y
float32 delta_z
string color


